### DISCUSSION TITLE
Sendmail: wrong parameters

### DISCUSSION THREAD

**Discussion by thygeL** [arendst/Tasmota#11319] [Q&A] [answered]
Why am getting "Wrong parameters" return in my esp8266-01 projekt ( tasmota 9.3.1)

I trying to build at temperature alarm, sending a email
I have scripting enabled, my ds18b20 sensor is running but when I try to send an email I get the message
``
My tasmota script
>D
temp1=0
>T
temp1=DS18B20#Temperature
if temp1>32
then
=>sendmail[*:*:*:*:*:xx@xx.xx:Temperatur alarm]*
endif

>m
email report %tstamp%
temperaturen er %temp1% grader
#
log output from console

19:15:21.031 Script: performs "sendmail[*:*:*:*:*:<my reciever emanil>:Temperatur alarm]*"
19:15:21.036 mailsize: 49
19:15:21.039 smtp.gmail.com - 465 - <my username> - <mypassword>
19:15:21.041 <sender address> - <reciever address> - Temperatur alarm - *
19:15:21.044 Connecting: smtp.gmail.com on port 465
19:15:21.457 220 smtp.gmail.com ESMTP e30sm1761015lfc.261 - gsmtp
19:15:21.519 EHLO 192.168.1.30
19:15:21.730 250-smtp.gmail.com at your service, [xx.x.xxx.xx]250-SIZE 35882577
250-8BITMIME
250-AUTH LOGIN PLAIN XOAUTH2 PLAIN-CLIENTTOKEN OAUTHBEARER XOAUTH
250-ENHANCEDSTATUSCODES
250-PIPELINING
250-CHUNKING
250 SMTPUTF8

19:15:21.792 AUTH LOGIN
19:15:21.797 334 VXNlcm5hbWU6
19:15:21.858 dGh5Z2VsYXJzZW5AZ21haWwuY29t
19:15:21.862 334 UGFzc3dvcmQ6
19:15:21.924 R3VsM3IwZDIw
19:15:22.126 
19:15:22.131 RSL: stat/tasmota_25F175/RESULT = {"Sendmail":"Wrong parameters"}
``

mv thyge

**Comment by barbudor** [comment, marked as answer]
Did you created an application specific password into your Google Account settings ?
You can't use your account password with GMail SMTP server

**Reply by thygeL** [reply]
No, but  I have now created a app-specific password in Google, but it gives me the same error

**Reply by barbudor** [reply]
In your above script, you haven't specified most of the parameters
You must specify them all or hard-code them in the user_config_override.h
Here is what I coded:
```
    #define EMAIL_USER "`mygoogleusername"
    #define EMAIL_PASSWORD "xxxxxxxxxx"
    #define EMAIL_FROM "<mygoogleusername@gmail.com>"
    #define EMAIL_SERVER "smtp.gmail.com"
    #define EMAIL_PORT 465
```

if you don't hardcode them, you must pass them on the command
beware that the FROM **MUST** include the brackets

**Reply by thygeL** [reply]
Yes, I have hardcoded in user_config_override.h
I got it to work with: 
app-specific password,
brackets aroung  from address 
brackets aroung To address 
thanks for the help

thyge

**Reply by Bebuccio** [reply]
I have used a customized version of "Tasmota 12.4.0 Peter" (#define USE_SENDMAIL).
With this version there is a difference between **ESP32** and **ES8266** version:
Using the version for ESP32 the email address must be written without the <>.
Using the ESP8266 version both email address (sender and receiver) **MUST** be surrounded by <>.

I.E. :
  for **ESP32** email address --> _user@server.com_
  for **ESP8266** email address --> _**<**user@server.com**>**_

**Comment by lalo-uy** [comment]
I send a PR for the documentation about brackets in the From for Gmail a
month ago.

**Comment by lalo-uy** [comment]
Your script will send a lot of emails, here you mine with hourly reminder

>D  25
> hr1=0
> et=0
> al=0

p:tm=25
> ***@***.***>"
> ***@***.***"
>
> >T
> et=DS18B20#Temperature
> if et>tm
> and al<1
> then
> al=1
> print  sending email
> =>sendmail [smtp.gmail.com:465:%fr%:dsaaambtsclkbol:<%fr%>:%to%:Temp
> Alarm]*
> endif
> if et<tm
> then
> al=0
> endif
>
> >S
> ; send every hour
> hr1=hours
> if chg[hr1]>0
> and al>0
> then
> print  sending email
> =>sendmail [smtp.gmail.com:465:%fr%:dsjakaaaatsclkbol:<%fr%>:%to%:Temp
> Alarm repeted]*
> endif
>
> >m
> Alarm at %tstamp%
> Temp now is %et% C
> #
>
> >W
> nm(1 30 1 tm \"Deg  \")
>

**Reply by thygeL** [reply]
Yes, thanks, i found out of that :-), but I have check on that part now

thyge
